blender_mcp_openai/dispatch.py

```
@@ -0,0 +1,164 @@
+            # Log and raise a structured error
+            logger.error(f"Error executing tool {tool_name}: {e}")
+            logger.debug(traceback.format_exc())
+            
+            raise ToolExecutionError(
+                tool_name,
+                str(e),
+                {
+                    "exception_type": type(e).__name__,
+                    "traceback": traceback.format_exc()
+                }
+            )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)